app/models/course/assessment/answer.rb

```diff
@@ -35,9 +35,13 @@ class Course::Assessment::Answer < ActiveRecord::Base
   def auto_grade!
     fail ArgumentError unless question.auto_gradable?
 
-    save!
-    create_auto_grading!
-    Course::Assessment::Answer::AutoGradingJob.new(auto_grading)
+    self.class.transaction do
+      save!
```

Comment:
Sorry ?  that's the reason why you are using transactions here. But why we want to propagate the errors in this transaction, who will handle those errors ?